meilisearch-release-changelog/v1.0.0.md

```diff
@@ -0,0 +1,121 @@
+
+This option is particular since it will not launch Meilisearhc but will generate a safe-to-use master key for Meilisearch and will exit without any error.
+
+Done in #3274 by @dureuill
+
+## Other improvements
+
+* Re-enable importing from dumps v1: you can now migrate from Meilisearch v0.20.0 and older to the latest version using a dump (#3170) @dureuill
+* Improve [`download-latest.sh`](https://github.com/meilisearch/meilisearch/blob/main/download-latest.sh) script: integrate apple-silicon binary (#3169) @curquiza
+* Clarify error message when the DB and engine versions are incompatible (#3279) @dureuill
+* Search speed improvements:
+  * synonyms (TBD)
+  * ranking rules (TDB)
+  * ... (TBD)
+
+# Breaking changes ⚠️
+
+Since we are releasing v1.0.0, this is the last time we are introducing breaking changes to this release before v2.0.0.
+
+## New CLI behavior
+
+* For security reasons, Meilisearch will now reject master keys that are less than 16 bytes (#3274) @dureuill
+* Remove `--max-index-size` and `--max-task-db` configuration options (#3278) @dureuill
+Indeed, these options did not include the real space Meilisearch takes on the disk and were not adapted to the user's needs.
+* Remove --disable-auto-batching CLI option (#3296) @loiclec
+This option was introduced in case of bug, and is not useful anymore since the feature is mature enough.
+* Rename dump command from `--dumps-dir` to `--dump-dir` (#3175) @dureuill
+* Remove `--snapshot-interval-sec` and `--schedule-snapshot` now accepts an optional integer value specifying the interval in seconds (#3281) @dureuill
+* The `--log-level` option does not take any string to filter the logs anymore, but only accepts the following arguments: `ERROR`, `WARN`, `INFO`, `DEBUG`, `TRACE` and `OFF` (#3293) @loiclec
+
+## New primary key inference
+
+When receiving the documents for the first time, and if the `primaryKey` parameter is not present in the request, Meilisearch tries to infer the primary key.
+
+In this case, Meilisearch will first try to find any key in the document containing `id`, like `puid` or `_id`. If only one key is detected, this key will be considered as the primary key. The changing behavior is about detecting multiple keys containing `id`.
+
+Old behavior: if Meilisearch finds multiple keys containing `id`, it will consider the primary key as the first field in the document.
+
+New behavior: to respect the JSON specification, if Meilisearch finds multiple keys containing `id`, it will now return an error. Indeed, there is no priority order in JSON documents.
+
+As previously, Meilisearch will throw an error is no key containing `id` are detected and if the `primaryKey` parameter is not present.
+
+Done in meilisearch/milli#752 and #??? (TBD) by @dureuill
+
+## Error handler changes
+
+* Add a new error `code`: `io_error` (used instead of `internal`) (#3263) @irevoire
+* Add a new error `code`: `too_many_open_files` (used instead of `internal`) (#3263) @irevoire
```

Comment:
```suggestion
* Add a new error `type`: `???` (#3263) @irevoire
    * `too_many_open_files` error `code` is linked to that error `type` instead of `internal`
    * `no_space_left_on_device` error `code` is linked to that error `type` instead of `internal`
    *`io_error` error `code` is linked to that error `type` instead of `internal`
```

Reply:
https://meilisearch.slack.com/archives/C04BRF22ZBP/p1672754110902429 Voted here 😃

Reply:
Waiting for the whole list of @irevoire before changing anything here 😇